web/src/components/wizard/config/ConfigurationStep.tsx

```diff
@@ -77,6 +83,103 @@ const ConfigurationStep: React.FC<ConfigurationStepProps> = ({ onNext }) => {
+        }
+      }
+    }
+    
+    return null;
+  };
+
+  // Helper function to find which group/tab contains a specific field
+  const findGroupForField = (fieldName: string): AppConfigGroup | null => {
+    if (!appConfig?.groups) return null;
+    
+    return appConfig.groups.find(group => 
+      group.items.some(item => item.name === fieldName)
+    ) || null;
+  };
+
+  // Helper function to focus on a field with tab switching support
+  const focusFieldWithTabSupport = (fieldItem: AppConfigItem): void => {
+    const targetGroup = findGroupForField(fieldItem.name);
+    
+    if (!targetGroup) {
+      console.warn(`Could not find group for field: ${fieldItem.name}`);
+      return;
+    }
+    
+    // Switch to the correct tab if field is in a different tab
+    if (targetGroup.name !== activeTab) {
+      setActiveTab(targetGroup.name);
+    }
+    
+    // Focus the field after a brief delay to ensure DOM updates complete
+    // This is especially important when switching tabs
+    const focusDelay = targetGroup.name !== activeTab ? FOCUS_DELAY_CROSS_TAB_MS : FOCUS_DELAY_SAME_TAB_MS;
+    setTimeout(() => {
+      let fieldElement: HTMLElement | null = null;
+      
+      // Special handling for file inputs - focus the upload button instead of hidden input
+      if (fieldItem.type === 'file') {
+        // For file inputs, focus the upload button using data-testid (follows pattern: file-input-{name}-button)
+        fieldElement = document.querySelector(`[data-testid="file-input-${fieldItem.name}-button"]`) as HTMLElement;
+      }
+      // Special handling for radio buttons - focus the first radio option  
+      else if (fieldItem.type === 'radio' && fieldItem.items && fieldItem.items.length > 0) {
+        // For radio buttons, focus the first radio option using its ID
+        fieldElement = document.getElementById(fieldItem.items[0].name);
+      }
+      // Special handling for checkboxes - focus the checkbox input
+      else if (fieldItem.type === 'bool') {
+        // For checkboxes, focus using the field ID
+        fieldElement = document.getElementById(fieldItem.name);
+      }
+      // Default case - use the field name as element ID
+      else {
+        fieldElement = document.getElementById(fieldItem.name);
+      }
+      
+      if (fieldElement) {
+        fieldElement.focus();
+        // Scroll the element into view to ensure it's visible
+        if (fieldElement.scrollIntoView) {
+          fieldElement.scrollIntoView({ behavior: 'smooth', block: 'center' });
+        }
+      } else {
+        console.warn(`Could not find DOM element for field: ${fieldItem.name} (type: ${fieldItem.type})`);
+      }
+    }, focusDelay);
+  };
+
+  // Helper function to parse server validation errors from API response
+  const parseServerErrors = (error: unknown): Record<string, string> => {
+    const fieldErrors: Record<string, string> = {};
+    
+    // Check if error has structured field errors
```

Comment:
Why not create a type for the server's validation errors? example:

- https://github.com/replicatedhq/embedded-cluster/blob/5163536446e16847e8bae0e46de8c271b7a4ec2f/web/src/components/wizard/setup/KubernetesSetupStep.tsx#L37-L39
- https://github.com/replicatedhq/embedded-cluster/blob/5163536446e16847e8bae0e46de8c271b7a4ec2f/web/src/components/wizard/setup/KubernetesSetupStep.tsx#L72